### PR TITLE
[SYCL][CUDA] Add tests for exceeding maximum number of work groups

### PR DESCRIPTION
**Pull request by t4c1**
Add tests for checking the error for exceeding maximum number of work groups.

Tests changes in https://github.com/intel/llvm/pull/4563.

### CODE REVIEW

**Review by alexbatashev — DISMISSED** [review]

**Review by alexbatashev — DISMISSED** [review]

File: SYCL/Basic/cuda_max_wgs_error.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+// REQUIRES: cuda
+
+#include <numeric>
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+
+const size_t lsize = 32;
+const size_t max_x = (1ull << 31ull) - 1ull;
+const size_t max_yz = 65535;
```
Comment (steffenlarsen) [inline SYCL/Basic/cuda_max_wgs_error.cpp]: Instead of using hard-coded values, would it make sense to use the [`max_work_groups` query extension](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_max_work_group_query.md)? Though I don't imagine these values will change on NVIDIA devices for a while, it should hopefully make it a little more robust and potentially make it easier to enable the test for other backends in the future.

Comment (t4c1) [reply SYCL/Basic/cuda_max_wgs_error.cpp]: done

**Review by steffenlarsen — APPROVED** [review]
Awesome, thanks for adding this! LGTM!

**Review by alexbatashev — APPROVED** [review]

**Comment by t4c1** [comment]
@steffenlarsen this can now be merged as well.
